### PR TITLE
Support older agent upgrades

### PR DESCRIPTION
**Pull request by wallyworld**
Juju 2.8 or earlier agents still look for series based agent metadata.
Restore some deleted "additional series" functionality and add a version check so that if the agent being upgraded is 2.8 or earlier, supply the extra upload tools args to cause the tools to be cloned for those series.

Also delete some unused code from core/series and refactor a couple of tests to allow more code deletion.

## QA steps

bootstrap a 2.8 bionic controller
add a focal machine
juju upgrade-controller --build-agent

check the upgrade completes
juju dump-db on controller model and check the toolsMetadata collection has info for focal and bionic

./main.sh upgrade

### CODE REVIEW

File: api/client.go [hunk 1]
```diff
@@ -559,8 +559,8 @@ func NewCharmDownloader(apiCaller base.APICaller) *downloader.Downloader {
 }
 
 // UploadTools uploads tools at the specified location to the API server over HTTPS.
-func (c *Client) UploadTools(r io.ReadSeeker, vers version.Binary) (tools.List, error) {
-	endpoint := fmt.Sprintf("/tools?binaryVersion=%s", vers)
+func (c *Client) UploadTools(r io.ReadSeeker, vers version.Binary, additionalSeries ...string) (tools.List, error) {
```
Comment (wallyworld) [inline api/client.go:562]: restored from 2.8

File: cmd/juju/commands/synctools_test.go [hunk 2]
```diff
@@ -263,7 +263,7 @@ func (s *syncToolsSuite) TestAPIAdapterUploadTools(c *gc.C) {
 	uploadToolsErr := errors.New("uh oh")
 	current := coretesting.CurrentVersion(c)
 	fake := fakeSyncToolsAPI{
-		uploadTools: func(r io.Reader, v version.Binary) (coretools.List, error) {
+		uploadTools: func(r io.Reader, v version.Binary, additionalSeries ...string) (coretools.List, error) {
```
Comment (wallyworld) [inline cmd/juju/commands/synctools_test.go:266]: these all restored from 2.8

**Review by ycliuhw — APPROVED** [review]

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

File: tests/suites/bootstrap/streams.sh [hunk 3]
```diff
@@ -21,7 +21,7 @@ run_simplestream_metadata() {
 
     file="${TEST_DIR}/test-bootstrap-stream.log"
     juju bootstrap "lxd" "${name}" \
-        --config agent-metadata-url="http://${ip_address}:8000/" \
+        --config agent-metadata-url="http://${ip_address}:8666/" \
```
Comment (SimonRichardson) [inline tests/suites/bootstrap/streams.sh:24]: You running Shoutcast/Winamp or something? Why the change?

**Comment by wallyworld** [comment]
[os based agent metadata part 2](https://trello.com/c/Bcspiq7H/1466-os-based-agent-metadata-part-2)
